File: Java/queue/meta_llama_llama_4_maverick_17b_128e_instruct.java

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.Random;
public class meta_llama_llama_4_maverick_17b_128e_instruct {
    public class Main {
        static class Queue {
            private int[] queue;
            private int front;
            private int rear;
            private int size;

            public Queue(int capacity) {
                queue = new int[capacity];
                front = 0;
                rear = 0;
                size = 0;
            }

            // Add an element
            public void enqueue(int item) {
                if (size == queue.length) {
                    // Handle queue overflow
                    return;
                }
                queue[rear] = item;
                rear = (rear + 1) % queue.length;
                size++;
            }

            // Remove an element
            public int dequeue() {
                if (size == 0) {
                    // Handle queue underflow
                    return -1;
                }
                int item = queue[front];
                front = (front + 1) % queue.length;
                size--;
                return item;
            }

            // Get the front element
            public int peek() {
                if (size == 0) {
                    // Handle queue underflow
                    return -1;
                }
                return queue[front];
            }

            // Check if the queue is empty
            public boolean isEmpty() {
                return size == 0;
            }

            // Check if the queue is full
            public boolean isFull() {
                return size == queue.length;
            }
        }
    }
    public static void main(String[] args) {
        Main.Queue queue = new Main.Queue(1000);

        try (BufferedReader br = new BufferedReader(new FileReader("../../inputs.txt"))) {
            String line;
            while ((line = br.readLine()) != null) {
                int value = Integer.parseInt(line.trim());
                queue.enqueue(value);
            }
        } catch (IOException e) {
            return;
        }

        // Desenfileirar e imprimir todos os elementos
        while (!queue.isEmpty()) {
            queue.dequeue();
        }

    }
}

File: C/queue/meta-llama-llama-4-maverick-17b-128e-instruct.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Queue {
    int* data;
    int front;
    int rear;
    int size;
    int capacity;
} Queue;

Queue* createQueue(int capacity) {
    Queue* queue = (Queue*)malloc(sizeof(Queue));
    queue->data = (int*)malloc(sizeof(int) * capacity);
    queue->front = 0;
    queue->rear = 0;
    queue->size = 0;
    queue->capacity = capacity;
    return queue;
}

int isEmpty(Queue* queue) {
    return queue->size == 0;
}

int isFull(Queue* queue) {
    return queue->size == queue->capacity;
}

void enqueue(Queue* queue, int value) {
    if (isFull(queue)) {
        // Handle queue full error
        return;
    }
    queue->data[queue->rear] = value;
    queue->rear = (queue->rear + 1) % queue->capacity;
    queue->size++;
}

int dequeue(Queue* queue) {
    if (isEmpty(queue)) {
        // Handle queue empty error
        return -1; // or some other sentinel value
    }
    int value = queue->data[queue->front];
    queue->front = (queue->front + 1) % queue->capacity;
    queue->size--;
    return value;
}

void testQueue() {
    Queue* queue = createQueue(1000);
    for (int i = 0; i < 1000; i++) {
        enqueue(queue, rand() % 1000);
    }
    for (int i = 0; i < 1000; i++) {
        dequeue(queue);
    }
    free(queue->data);
    free(queue);
}

int main() {
    FILE *file = fopen("../../inputs.txt", "r");
    if (!file) {
        return 1;
    }

    Queue *q = createQueue(1000); 
    if (!q) {
        fclose(file);
        return 1;
    }

    int num, count = 0;
    while (count < 1000 && fscanf(file, "%d", &num) == 1) {
        enqueue(q, num);
        count++;
    }
    fclose(file);

    free(q); 
    return 0;
}

